geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -383,6 +389,51 @@ long zrevrank(byte[] member) {
     return null;
   }
 
+  long zunionstore(RegionProvider regionProvider, RedisKey key, List<ZKeyWeight> keyWeights,
+      ZAggregator aggregator) {
+    for (ZKeyWeight keyWeight : keyWeights) {
+      RedisSortedSet set =
+          regionProvider.getTypedRedisData(REDIS_SORTED_SET, keyWeight.getKey(), false);
+      if (set == NULL_REDIS_SORTED_SET) {
+        continue;
+      }
+      double weight = keyWeight.getWeight();
+
+      for (AbstractOrderedSetEntry entry : set.members.values()) {
+        OrderedSetEntry existingValue = members.get(entry.member);
+        if (existingValue == null) {
+          byte[] score;
+          // Redis math and Java math are different when handling infinity. Specifically:
+          // Java: INFINITY * 0 = NaN
+          // Redis: INFINITY * 0 = 0
+          if (weight == 0) {
+            score = bZERO;
+          } else if (weight == 1) {
+            score = entry.getScoreBytes();
+          } else {
+            double newScore = entry.score * weight;
+            if (Double.isNaN(newScore)) {
+              score = entry.getScoreBytes();
+            } else {
+              score = Coder.doubleToBytes(entry.score * weight);
+            }
+          }
+          members.put(entry.member, new OrderedSetEntry(entry.member, score));
+          continue;
+        }
+
+        existingValue.updateScore(aggregator.getFunction().apply(existingValue.score,
+            entry.score * weight));
```

Comment:
you had a comment up above about how redis and java do math differently. Why can this code just do java math?